test/functional/rpc_getblockstats.py

```diff
@@ -40,40 +40,47 @@ def get_stats(self):
         return [self.nodes[0].getblockstats(hash_or_height=self.start_height + i) for i in range(self.max_stat_pos+1)]
 
     def generate_test_data(self, filename):
+        node = self.nodes[0]
         mocktime = 1525107225
-        self.nodes[0].setmocktime(mocktime)
-        self.nodes[0].generate(101)
+        node.setmocktime(mocktime)
+        node.generate(101)
 
-        address = self.nodes[0].getnewaddress('', 'bech32')
-        self.nodes[0].sendtoaddress(address=address, amount=10, subtractfeefromamount=True)
-        self.nodes[0].sendtoaddress(address=address, amount=10, subtractfeefromamount=True)
-        self.nodes[0].generate(1)
-        self.sync_all()
+        pk1 = node.getaddressinfo(node.getnewaddress())['pubkey']
+        pk2 = node.getaddressinfo(node.getnewaddress())['pubkey']
+
+        nested_p2wpkh_address = self.nodes[0].getnewaddress('', 'bech32')
```

Comment:
Can you move these changes to the first commit (_Implement SegWit related stats in getblockstats_)? That way the new statistics get tested in the same commit that they're introduced in.

Ideally we want each new statistic to be tested in the commit that introduces it. For _Implement SegWit related stats in getblockstats_, those statistics are:

- nested_p2wpkh_outputs_spent
- nested_p2wsh_outputs_spent
- native_p2wpkh_outputs_spent
- native_p2wsh_outputs_spent
- txs_spending_nested_p2wpkh_outputs
- txs_spending_nested_p2wsh_outputs
- txs_spending_native_p2wpkh_outputs
- txs_spending_native_p2wsh_outputs
- new_p2wpkh_outputs
- new_p2wsh_outputs
- txs_creating_p2wpkh_outputs
- txs_creating_p2wsh_outputs

so you want to create an address for each type (as you do here), and make sure funds are sent to those addresses and subsequently spent.

Reply:
I think the only type of transaction you've missed here is spending from a p2sh-p2wpkh. If you add that, I think you've triggered all the stats.

Reply:
p2sh-p2wpkh was handled previously with:

`nested_p2wpkh_address = node.get_deterministic_priv_key().address`

Looks like you missed that in copying over from my branch. I will amend.

Reply:
`get_deterministic_priv_key()` doesn't work here because that function didn't exist when this branch was created, which is why I changed this.

I made a mistake though. We need to change:

`nested_p2wpkh_address = self.nodes[0].getnewaddress('', 'bech32')`

to be:

`nested_p2wpkh_address = self.nodes[0].getnewaddress('', 'p2sh-segwit')`